sympy/matrices/matrices.py

```diff
@@ -2734,7 +2737,7 @@ def rref(self, iszerofunc=_iszero, simplify=False):
             pivot += 1
         return self._new(r), pivotlist
 
-    def rank(self, iszerofunc=_iszero, simplify=False):
+    def rank(self, iszerofunc=_iszero, simplify=True):
```

Comment:
This was discussed in the comments to the issue. Though it appears to lead to correct results, always calling simplify by default adds a heavy overhead to the computation. So it seems that this line should be left as is and another way should be found to alert the user in case iszerofunc returns None.